### PR TITLE
Create and use a decorator to wrap aiopg and pyscop2 erros as connect…

### LINKED ISSUE #140: Wrap errors in aiopg connector
Currently, database errors are raised from `psycopg2` or `aiopg`. However, we want to use our own errors, so that we have less dependancy with those librairies.

We need to catch all psyopg2/aiopg errors at the source so that we use our own custom-defined errors, with a `from` argument to not lose the base error.

This can probably be done inside a decorator that we apply on all `aiopg_connector` functions.

### PR DESCRIPTION
**Pull request by SBillion**
Closes #140 

### Successful PR Checklist:
- [x] Tests
- [x] Documentation (optionally: [run spell checking](https://github.com/peopledoc/procrastinate/blob/master/CONTRIBUTING.rst#build-the-documentation))
- [x] Had a good time contributing? (if not, feel free to give some feedback)

This PR adds a decorator to wrap psycopg2 and aiopg errors as connector exceptions. This decorator is expected to be used on coroutine functions only. It has been used on PostgresConnector methods doing psycopg2 or aiopg potential errors

### CODE REVIEW

File: procrastinate/aiopg_connector.py [hunk 1]
```diff
@@ -1,18 +1,34 @@
 import asyncio
+import contextlib
```
Comment (thomasperrot) [inline procrastinate/aiopg_connector.py:2]: Cet import n'est plus nécessaire :)

**Comment by ewjoachim** [comment]
Ready for review !

**Comment by ewjoachim** [comment]
:o coverage has dropped below 100%

I'll fix that elsewhere.

File: procrastinate/exceptions.py [hunk 2]
```diff
@@ -23,4 +30,19 @@ def __init__(self, scheduled_at: datetime.datetime):
 
 
 class PoolAlreadySet(ProcrastinateException):
+    """
+    Indicates a call on connector.set_pool() was done but the
+    pool already had a set. Chaging the pool of a connector is not
```
Comment (CorBott) [inline procrastinate/exceptions.py:35]:
```suggestion
    pool already had a set. Changing the pool of a connector is not
```

File: procrastinate/aiopg_connector.py [hunk 3]
```diff
@@ -12,6 +14,25 @@
...
+    """
+
+    @functools.wraps(coro)
+    async def wrapped(*args, **kwargs):
+        try:
+            return await coro(*args, **kwargs)
+        except psycopg2.Error as exc:
+            raise exceptions.ConnectorException from exc
+
+    wrapped._exceptions_wrapped = True  # type: ignore
```
Comment (elemoine) [inline procrastinate/aiopg_connector.py:33]: Is this only for the test? If so I think this deserves a comment in the code.

Comment (ewjoachim) [inline procrastinate/aiopg_connector.py:33]:
```suggestion
    # Attaching a custom attribute to ease testability and make the
    # decorator more introspectable
    wrapped._exceptions_wrapped = True  # type: ignore
```

**Pull request merged**